compiler/resolution/virtualDispatch.cpp

```diff
@@ -527,19 +528,21 @@ static bool isSubType(Type* sub, Type* super) {
 }
 
 static bool isOverrideableMethod(FnSymbol* fn) {
-  AggregateType* receiver = fn->getReceiver();
-  if (receiver)
-    if (AggregateType* ct = toAggregateType(receiver->getValType()))
-      if (isClass(ct))
-        return fn->name != astrInit &&
-               !fn->hasFlag(FLAG_WRAPPER) &&
-               !fn->hasFlag(FLAG_NO_PARENS) &&
-                fn->retTag != RET_PARAM &&
-                fn->retTag != RET_TYPE;
+  if (AggregateType* at = getReceiverClassType(fn)) {
+    INT_ASSERT(at->isClass()); 
+    return fn->name != astrInit &&
+           !fn->hasFlag(FLAG_WRAPPER) &&
+           !fn->hasFlag(FLAG_NO_PARENS) &&
+            fn->retTag != RET_PARAM &&
+            fn->retTag != RET_TYPE;
```

Comment:
I'm not so confident that you want to have RET_PARAM and RET_TYPE apply to your type methods. Try making some tests with e.g. `proc type method() type { }` and `proc type method() param { }`

Reply:
Thanks, I'll add a few more tests.